apps/arena/lib/arena/game_updater.ex

```diff
@@ -824,4 +934,23 @@ defmodule Arena.GameUpdater do
       end
     end)
   end
+
+  defp handle_pools(game_state) do
+    Enum.each(game_state.players, fn {player_id, player} ->
+      colliding_pools =
+        game_state.pools
+        |> Map.filter(fn {pool_id, _pool} -> pool_id in player.collides_with end)
+
+      Enum.each(game_state.pools, fn {pool_id, pool} ->
+        if Map.has_key?(colliding_pools, pool_id) do
```

Comment:
You don't need the `colliding_pools` variable, you can replace this `if` like

```elixir
if pool_id in player.collides_with do
  ...
else
  ...
end
```

Reply:
i made the change! the colliding pools was only for this check